clients/venice-push-job/src/main/java/com/linkedin/venice/hadoop/input/kafka/KafkaInputRecordReader.java

```diff
@@ -292,75 +223,27 @@ public KafkaInputMapperValue createValue() {
 
   @Override
   public long getPos() {
-    return recordsReadSoFar - 1; // Return the last read record logical position
+    // Return the last-read logical position
+    return pubSubSplitIterator.recordsRead() - 1;
   }
 
   @Override
   public void close() {
-    if (ownedConsumer) {
-      this.consumer.close();
+    // Close only if we own the consumer; otherwise the caller manages its lifecycle.
+    if (ownedConsumer && pubSubSplitIterator != null) {
+      pubSubSplitIterator.close();
```

Comment:
why are we not closing consumer?

Reply:
If this class created consumer then it is indeed closed in split iterator. But if it was injected then it's caller's responsibility to close it

```
  @Override
  public void close() {
    if (!closed) {
      pubSubConsumer.close();
      closed = true;
      LOGGER.info(
          "PubSubSplitIterator closed for {}. read={}, delivered={}, skipped={}",
          topicPartition,
          readSoFar,
          recordsDelivered,
          controlMessagesSkipped);
    }
  }
```